docs/articles_en/documentation/openvino-extensibility/custom-openvino-operations.rst

```diff
@@ -7,30 +7,64 @@ Custom OpenVINO Operations
+      Follow the steps below to add a simple custom operation:
 
-2. Implement default constructor and constructors that optionally take the operation inputs and attributes as parameters.
+      1. Define the ``__init__`` method to initialize the class with inputs and attributes.
 
-3. Override the shape inference method ``validate_and_infer_types``. This method is called multiple times during graph manipulations to determine the shapes and element types of the operations outputs. To access the input shapes and input element types, use the ``get_input_partial_shape()`` and ``get_input_element_type()`` methods of ``ov::Node``. Set the inferred shape and element type of the output using ``set_output_type``.
+      2. Override the shape inference method ``validate_and_infer_types``. This method is called multiple times during graph manipulations to determine the shapes and element types of the operations outputs. To access the input shapes and input element types, use the ``get_input_partial_shape()`` and ``get_input_element_type()`` methods of ``openvino.Node``. Set the inferred shape and element type of the output using ``set_output_type``.
 
-4. Override the ``clone_with_new_inputs`` method, which enables graph manipulation routines to create copies of this operation and connect it to different nodes during optimization.
+      3. Override the ``visit_attributes`` method, which enables serialization and deserialization of operation attributes. An ``AttributeVisitor`` is passed to the method, and the implementation is expected to walk over all the attributes in the op using the type-aware ``on_attribute`` helper. Helpers are already implemented for standard types like ``int``, ``float``, ``bool``, ``vector``, and for existing OpenVINO defined types.
 
-5. Override the ``visit_attributes`` method, which enables serialization and deserialization of operation attributes. An ``AttributeVisitor`` is passed to the method, and the implementation is expected to walk over all the attributes in the op using the type-aware ``on_attribute`` helper. Helpers are already implemented for standard C++ types like ``int64_t``, ``float``, ``bool``, ``vector``, and for existing OpenVINO defined types.
+      There are some methods that can also be overridden if needed:
+
+      4. Override the ``clone_with_new_inputs`` method, which enables graph manipulation routines to create copies of this operation and connect it to different nodes during optimization.
+
+      5. Override ``evaluate``, which is an optional method that enables fallback of some devices to this implementation and the application of constant folding if there is a custom operation on the constant branch. If your operation contains ``evaluate`` method you also need to override the ``has_evaluate`` method, this method allows to get information about availability of ``evaluate`` method for the operation.
```

Comment:
Well, `evaluate` is the main reason why we need to define a custom Python op. Exactly for `fallback of some devices to this implementation`. So we literally want to run this op as a part of the graph, and this is the main purpose of the recent advances in Python Op. So, technically, it is optional, but for the main use case -- it is mandatory.

Reply:
Please use a more direct language to express meaning of this Python op: a way to run an arbitrary Python code as part of OpenVINO model graph. Highlight that this is available for CPU only.

Reply:
@slyalin `Highlight that this is available for CPU only.`
Am I right that it is correct also for C++ API?

Reply:
Correct. But it was a feature requested for GPU for a long time. So it is one of the things that can be enabled automatically for all devices, but the code itself still will be executed on the CPU. But I hope we will move further and enable, for example, Python custom operation implemented in Triton and run them natively on GPU as well.